src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -403,22 +403,88 @@ private static Version GetMSBuildVersionInPath()
         /// <returns>The msbuild directory.</returns>
         public static string GetMsbuildDirectory(string userVersion, IConsole console)
         {
-            List<Toolset> installedToolsets;
-            using (var projectCollection = new ProjectCollection())
+            // Try to find msbuild for mono from hard code path.
+            if (RuntimeEnvironmentHelper.IsMono)
             {
-                installedToolsets = projectCollection.Toolsets.OrderByDescending(
-                    toolset => SafeParseVersion(toolset.ToolsVersion)).ToList();
+                if (userVersion != null)
+                {
+                    switch (userVersion)
+                    {
+                        case "14.1": return CommandLineConstants.MsbuildPathOnMac14;
+                        case "15":
+                        case "15.0": return CommandLineConstants.MsbuildPathOnMac15;
+                    }
+                }
+                else
+                {
+                    var path = new[] { new Tuple<string, string>("15.0", CommandLineConstants.MsbuildPathOnMac15),
+                       new Tuple<string, string>("14.1", CommandLineConstants.MsbuildPathOnMac14) }.FirstOrDefault(p => Directory.Exists(p.Item2));
+
+                    if (path != null)
```

Comment:
No, it can be false, if the two directory don't exist, it will return default value of reference, that's null